src/wallet/rpcwallet.cpp

```diff
@@ -3706,9 +3706,18 @@ UniValue z_getbalance(const UniValue& params, bool fHelp)
             nBalance = getBalanceZaddr(addr, nMinDepth, INT_MAX, false);
         },
         [&](const libzcash::UnifiedAddress& addr) {
-             throw JSONRPCError(
-                     RPC_INVALID_ADDRESS_OR_KEY,
-                     "Unified addresses are not yet supported for z_getbalance.");
+            auto selector = pwalletMain->ZTXOSelectorForAddress(addr, true);
+            auto spendableInputs = pwalletMain->FindSpendableInputs(selector.value(), true, nMinDepth);
```

Comment:
We are restricting this to supporting UAs for which we know the spending key (via `ZTXOSelectorForAddress(_, true)`) which is the correct behaviour here (we want to show the account balance here, not the balance restricted to that address's receiver types). As such, the `selector.value()` here will throw an exception if the caller provides a different UA. This will manifest as an unexpected exception being returned, instead of `JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, _)` which is what we should return in this case.
```suggestion
            auto selector = pwalletMain->ZTXOSelectorForAddress(addr, true);
            if (!selector.has_value()) {
                throw JSONRPCError(
                    RPC_INVALID_ADDRESS_OR_KEY,
                    "Unified address does not correspond to an account in the wallet");
            }
            auto spendableInputs = pwalletMain->FindSpendableInputs(selector.value(), true, nMinDepth);
```